doc/api/cli.md

```diff
@@ -11,16 +11,42 @@ To view this documentation as a manual page in a terminal, run `man node`.
 
 ## Synopsis
 
-`node [options] [V8 options] [script.js | -e "script" | -] [--] [arguments]`
+`node [options] [V8 options] [<program-entry-point> | -e "script" | -] [--] [arguments]`
 
-`node inspect [script.js | -e "script" | <host>:<port>] …`
+`node inspect [<program-entry-point> | -e "script" | <host>:<port>] …`
 
 `node --v8-options`
 
 Execute without arguments to start the [REPL][].
 
 For more info about `node inspect`, see the [debugger][] documentation.
 
+## Program entry point
+
+The program entry point is a specifier-like string. If the string is not an
+absolute path, it's resolved as a relative path from the current working
+directory. That path is then resolved by [CommonJS][] module loader. If no
+corresponding file is found, an error is thrown.
+
+If a file is found, its path will be passed to the [ECMAScript module loader][]
+under any of the following conditions:
+
+* The file has an `.mjs` extension.
+* The file nearest parent `package.json` file
```

Comment:
Sorry if this is rehashing prior threads, but is this true if i pass a .cjs file, or an extensionless one? Surely the ESM loader only is used for an ESM entry point.

Reply:
No it's always used, even for `.cjs` files (for `.cjs`, because it's associated with `'commonjs'` format, the ES module loader would pass its content through `es-module-lexer` and finally back to the CJS module loader). It can be illustrated by trying to load an extension-less file:

```sh
mkdir repro
cd repro
echo 'console.log(this === undefined ? "ESM" : "CJS")' > file
node file # CJS
echo '{"type":"module"}' > package.json
node file # ERR_UNKNOWN_FILE_EXTENSION
```

> Surely the ESM loader only is used for an ESM entry point.

Well we can't know what format it is until we actually resolved the specifier (or we would need some kind of import assertion for the CLI entry point 🙃).

Reply:
It seems to me like we could - if it's a relative path, then either the specifier has an `.mjs` extension, or it has a `.js` extension and the closest package.json file within the package boundary has `type` "module", or else it's CJS; if it's a bare specifier, then we'd use normal resolution to figure that out.

Reply:
For a `.cjs` file, when there is no custom loaders, it does sound like some unnecessary extra work to go through the ESM loader, indeed. The fact that you didn't notice until I opened the recent PRs could indicate that it may be negligible after all 😅
Anyway, this PR is focusing on documenting the current behavior of Node.js, even if said behavior is not in its final form or ideal state.

Reply:
I didn’t notice because i never use type module :-p

Reply:
I was wrong on this, `.cjs` entry points do not go through ESM loader, regardless of any neighbour `package.json`:
https://github.com/nodejs/node/blob/775bfd15799e57ae8b951b51cea7e86f10bfdc1d/lib/internal/modules/run_main.js#L41-L42
I must must say that makes much more sense, I don't know how I missed that earlier.